library/std/src/sys/unix/ext/fs.rs

```diff
@@ -345,6 +345,33 @@ pub trait OpenOptionsExt {
     /// ```
     #[stable(feature = "open_options_ext", since = "1.10.0")]
     fn custom_flags(&mut self, flags: i32) -> &mut Self;
+
+    /// Get the flags of this OpenOptions as libc::c_int.
+    ///
+    /// This method allows the reuse of the OpenOptions as flags argument for `libc::open()`.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # #![feature(rustc_private)]
+    /// extern crate libc;
+    /// use std::ffi::CString;
+    /// use std::fs::OpenOptions;
+    /// use std::os::unix::fs::OpenOptionsExt;
+    ///
+    /// # fn main() {
```

Comment:
I believe `fn main() {` isn't needed here.

Reply:
I've copied it from the `custom_flags` docs... shall I remove it also from there and from `mode`?

Reply:
We could probably remove that in a follow up patch. Do you want to do it?

Reply:
sure, I'll do it!